content/blog/2017/2017-01-18-refactoring-conditional-to-pipeline.adoc

```diff
@@ -0,0 +1,475 @@
+and some provide information that is simply not exposed in pipeline yet.
+So, determining how to migrate tokens needs to be done on case-by-case basis.
+
+Let's look at a few examples.
+
+==== "FILE" token
+
+*Expands to the contents of a file. The file path is relative to the build workspace root.*
+
+_${FILE,path="PATH"}_
+
+This token maps directly to the `readFile` step.
+The only difference is the file path for `readFile` is relative to the
+current working directory on the agent, but that is the workspace root by default.
+No problem.
+
+.Pipeline
+[pipeline]
+----
+// Declarative //
+when {
+    expression { return readFile('pom.xml').contains('mycomponent') }
+}
+steps {
+    /* step */
+}
+
+// Script //
+if (readFile('pom.xml').contains('mycomponent')) {
+    /* step */
+}
+----
+
+
+==== GIT_BRANCH
+
+*Expands to the name of the branch that was built.*
+
+*Parameters* (descriptions omitted): _all_, _fullName_.
+
+This information may or may not be exposed in Pipeline.  If you're using
+multibranch pipeline plugin `env.BRANCH_NAME` give similar basic information, but doesn't
+offer the parameters.  There are also
+link:https://issues.jenkins-ci.org/browse/JENKINS-35230[several]
+link:https://issues.jenkins-ci.org/browse/JENKINS-24141[issues]
+filed around `GIT_*` tokens in Pipeline.
+Until they are addressed fully, we can follow the pattern shown in
+link:https://github.com/jenkinsci/pipeline-examples/blob/master/pipeline-examples/gitcommit/gitcommit.groovy[pipeline-examples],
+executing a shell to get the information we need.
+
+
+.Pipeline
+[source, groovy]
+----
+GIT_BRANCH = sh(returnStdout: true, script: 'git rev-parse --abbrev-ref HEAD').trim()
+----
+
+==== CHANGES_SINCE_LAST_SUCCESS
+
+*Displays the changes since the last successful build.*
+
+*Parameters* (descriptions omitted):
+_reverse_, _format_, _changesFormat_, _showPaths_, _pathFormat_,
+_showDependencies_, _dateFormat_, _regex_, _replace_, _default_.
+
+Not only is the information provided by this token not exposed in Pipeline,
+the token has ten optional parameters, including format strings and regular expression
+searches. There are some examples,
+link:http://stackoverflow.com/questions/38084806/how-to-get-the-changes-since-the-last-successful-build-in-jenkins-pipeline[this answer on stackoverflow.com],
+that show how to get similar information in Pipeline,
+but nothing comes close to functional parity.
+Translating this token is beyond the scope of this post.
+
+== Slightly More Complex Example
+
+Let's do one more example that shows some of these conditions and tokens.
+This time we'll perform different build steps depending on what branch we're building.
+
+=== Freestyle
+
+Here's the configuration for Freestyle version
+(It's pretty long, click <<longer-pipeline, here>> to skip down to the Pipeline version):
+
+image::/images/post-images/2017-01-17/freestyle-conditional-long-full.png[Longer Freestyle Job, role="center", width=800]
+
```

Comment:
A one-sentence description of what the above image does would be nice-to-have here.